clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -287,9 +287,8 @@ std::set<ConstraintVariable *>
           // Taking the address of a dereference is a NoOp, so the constraint
           // vars for the subexpression can be passed through.
           return getExprConstraintVars(SubUO->getSubExpr());
-        // TODO: this should also work for array subscript (issue #51), but it break some regression tests.
-        //} else if (ArraySubscriptExpr *ASE = dyn_cast<ArraySubscriptExpr>(UOExpr)) {
-        //  return getExprConstraintVars(ASE->getBase());
+        } else if (ArraySubscriptExpr *ASE = dyn_cast<ArraySubscriptExpr>(UOExpr)) {
+          if(!AllTypes) return getExprConstraintVars(ASE->getBase());
         } else {
```

Comment:
But what happens when `AllTypes` is *not* true? There's a fallthrough here to ... what? Can you explain why whatever happens is going to be correct?